code/modules/mob/living/simple_animal/hostile/onestar_drone.dm

```diff
@@ -103,3 +104,49 @@
 	melee_damage_lower = 7
 	melee_damage_upper = 15
 	rarity_value = 39.66
+	move_to_delay = 5
+	
+
+/mob/living/simple_animal/hostile/onestar_custodian/engineer/Life()
+	if(!stasis && !AI_inactive)
+		if(!stat)
+			switch(stance)
+				if(HOSTILE_STANCE_IDLE)
+					target_mob = FindTarget()
+
+				if(HOSTILE_STANCE_ATTACK)
+					if(destroy_surroundings)
+						DestroySurroundings()
+					MoveToTarget()
+				if(HOSTILE_STANCE_ATTACKING)
+					if(destroy_surroundings)
+						DestroySurroundings()
+					MoveToTarget()
```

Comment:
Lack of parent call can and will cause issues.
`SEND_SIGNAL_OLD(src, COMSIG_MOB_LIFE)`, `handle_stuff()` procs, and health checks are all missing.